tests/core/full_node/test_block_height_map.py

```diff
@@ -374,6 +375,76 @@ async def test_rollback2(self, tmp_dir: Path, db_version: int) -> None:
             with pytest.raises(KeyError) as _:
                 height_map.get_ses(uint32(8))
 
+    @pytest.mark.asyncio
+    async def test_cache_file_nothing_to_write(self, tmp_dir: Path, db_version: int) -> None:
+        # This is a test where the height-to-hash data is entirely used from
+        # the cache file and there is nothing to write in that file as a result.
+        async with DBConnection(db_version) as db_wrapper:
+            await setup_db(db_wrapper)
+            await setup_chain(db_wrapper, 10000, ses_every=20)
+            await BlockHeightMap.create(tmp_dir, db_wrapper)
+            # To ensure we're actually loading from cache, and not the DB, clear
+            # the table.
+            async with db_wrapper.writer_maybe_transaction() as conn:
+                await conn.execute("DROP TABLE full_blocks")
+            await setup_db(db_wrapper)
+            await setup_chain(db_wrapper, 10000, ses_every=20, start_height=9970)
+            # At this point we should have a proper cache file
+            heights_stat = (tmp_dir / "height-to-hash").stat()
+            assert heights_stat.st_size == (10000 + 1) * 32
+            await BlockHeightMap.create(tmp_dir, db_wrapper)
+            # Make sure we didn't alter the cache (nothing to write)
+            new_heights_stat = (tmp_dir / "height-to-hash").stat()
+            assert new_heights_stat.st_mtime == heights_stat.st_mtime
+            assert new_heights_stat.st_size == heights_stat.st_size
```

Comment:
if the timestamp only has seconds resolution, this wouldn't tell you if the file was updated. You would a sleep somewhere probably

Reply:
As the intent here is to check the size (and content) match, I removed this additional time check.

Reply:
you don't check the content now though. did you intend to?